src/rpc/mining.cpp

```diff
@@ -251,7 +251,7 @@ static UniValue getmininginfo(const JSONRPCRequest& request)
     if (BlockAssembler::m_last_block_num_txs) obj.pushKV("currentblocktx", *BlockAssembler::m_last_block_num_txs);
     obj.pushKV("difficulty",       (double)GetDifficulty(::ChainActive().Tip()));
     obj.pushKV("networkhashps",    getnetworkhashps(request));
-    obj.pushKV("pooledtx",         (uint64_t)mempool.size());
+    obj.pushKV("pooledtx",         (uint64_t)EnsureMemPool().size());
```

Comment:
I think I'd prefer to have this check explicitly called at the top of this function rather than hidden in a univalue value assignment.

Same is true for other changes in this file. I think it's better to be explicit at the top of the function that a mempool is going to be required, and fail early if it isn't available.

Reply:
Done: Moved `EnsureMemPool` right after parsing the args where applicable.